src/math/lp/lar_solver.cpp

```diff
@@ -797,7 +797,17 @@ namespace lp {
         lp_assert(((stats().m_make_feasible% 100) != 0) || m_status != lp_status::OPTIMAL || all_constraints_hold());
     }
 
-
+    bool lar_solver::status_is_feasible(lp_status st) {
```

Comment:
just call it is_feasible() and use get_status() internally.

Reply:
The use case is here:

https://github.com/Z3Prover/z3/blob/0242566792d858275b39175936ad35e1faa22c63/src/smt/theory_lra.cpp#L1532

The new code would be

```
  if (!lp().is_feasible() || lp().has_changed_columns()) 
```
or even the has_changed_columns() really implies that the solver is not in the feasible state.